src/components/app/details/appDetails/DeploymentStatusModal.tsx

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import { Drawer } from '../../../common';
+import { ReactComponent as Close } from '../../../../assets/icons/ic-close.svg';
+import { ReactComponent as Progressing } from '../../../../assets/icons/appstatus/progressing-rotating.svg';
+import { ReactComponent as Waiting } from '../../../../assets/icons/ic-clock.svg';
+import { ReactComponent as Failed } from '../../../../assets/icons/appstatus/ic-appstatus-failed.svg';
+import { ReactComponent as Success } from '../../../../assets/icons/ic-outline-check.svg';
+import { ReactComponent as Unknown } from '../../../../assets/icons/appstatus/unknown.svg';
+import Uncheck from '../../../../assets/img/ic-success@2x.png';
+
+export function DeploymentStatusModal({ appName, environmentName, deploymentStatus, close }) {
+    let gitPushStep = deploymentStatus?.gitPushStep?.status.toLowerCase();
+    let gitPullStep = deploymentStatus?.gitPullStep?.status.toLowerCase();
+    let configApplyStep = deploymentStatus?.configApplyStep?.status.toLowerCase();
+    let k8sDeploy = deploymentStatus?.k8sDeploy?.status.toLowerCase();
+    let lastDeploymentStatus = deploymentStatus?.lastDeploymentStatus?.toLowerCase() || "";
+    let message = {
+        error: "Error"
+    }
+
+    return <Drawer position="right" width="800px" onClose={close}>
+        <div className="app-details-status-modal bcn-0" onClick={(e) => e.stopPropagation()}>
+            <div className="pl-20 pr-20 pt-12 pb-12 flex flex-align-center flex-justify" style={{ borderBottom: "1px solid #d0d4d9" }}>
+                <div>
+                    <h2 className="fs-16 lh-1-5 fw-6 m-0">Deployment status: {appName} / {environmentName}</h2>
+                    <p className={`m-0 text-uppercase app-summary__status-name fs-12 fw-6 f-${lastDeploymentStatus}`}>
+                        {lastDeploymentStatus}
+                    </p>
+                </div>
+                <button type="button" className="transparent flex icon-dim-24" onClick={close}>
+                    <Close className="icon-dim-24" />
+                </button>
+            </div>
+            <div className="p-20">
+                <div className="position-rel add-line add-line--git-push">
```

Comment:
each of these steps have same structure...we can extract out the component

Reply:
Done.